nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteStreamCommand.java

```diff
@@ -419,21 +422,27 @@ public void onTrigger(ProcessContext context, final ProcessSession session) thro
                 }
                 return 0;
             });
-            for ( final PropertyDescriptor descriptor : propertyDescriptors) {
+
+            for (final PropertyDescriptor descriptor : propertyDescriptors) {
+                if (descriptor.isSensitive()) {
+                    String maskedValue = IntStream.rangeClosed(0, MASKED_VALUE_LENGTH).mapToObj(i -> "*").collect(Collectors.joining());
+                    argumentAttributeValue.add(maskedValue);
+                } else {
+                    argumentAttributeValue.add(context.getProperty(descriptor.getName()).evaluateAttributeExpressions(inputFlowFile).getValue());
+                }
                 args.add(context.getProperty(descriptor.getName()).evaluateAttributeExpressions(inputFlowFile).getValue());
```

Comment:
The evaluated value of `context.getProperty(...)` should be extracted to a variable and reused.

Reply:
sure i will assign value to variable and reuse